src/app.ts

```diff
@@ -86,6 +98,101 @@ const cachingMiddleware = apicache.options({
   }
 }).middleware;
 
+const SAML_NAMESPACE = {
+  ASSERTION: "urn:oasis:names:tc:SAML:2.0:assertion",
+  PROTOCOL: "urn:oasis:names:tc:SAML:2.0:protocol",
+  XMLDSIG: "http://www.w3.org/2000/09/xmldsig#"
+};
+
+const getFiscalNumberFromPayload = (doc: Document): Option<string> => {
+  return fromNullable(
+    doc.getElementsByTagNameNS(SAML_NAMESPACE.ASSERTION, "Attribute")
+  )
+    .mapNullable(
+      collection =>
+        Array.from(collection).filter(
+          elem => elem.getAttribute("Name") === "fiscalNumber"
+        )[0]
+    )
+    .mapNullable(_ => _.textContent?.trim());
+};
+
+const getRequestIDFromPayload = (
+  doc: Document,
+  tagName: string,
+  attrName: string
+): Option<string> => {
+  return fromNullable(
+    doc.getElementsByTagNameNS(SAML_NAMESPACE.PROTOCOL, tagName).item(0)
+  ).chain(element =>
+    fromEither(NonEmptyString.decode(element.getAttribute(attrName)))
+  );
+};
+
+const callback = (
+  sourceIp: string | null,
+  payload: string,
+  payloadType: "REQUEST" | "RESPONSE"
+): void => {
+  const { QueueClient } = require("@azure/storage-queue");
+  const maybeXmlPayload = tryCatch(() =>
+    new DOMParser().parseFromString(payload, "text/xml")
+  );
+  if (isNone(maybeXmlPayload)) {
+    log.error(
+      `Spid Log callback| ERROR | Impossible to parse SPID XML Payload`
+    );
+    return void 0;
+  }
+  const idParams =
+    payloadType === "REQUEST"
+      ? { tagName: "AuthnRequest", attrName: "ID" }
+      : { tagName: "Response", attrName: "InResponseTo" };
+  const tId = getRequestIDFromPayload(
+    maybeXmlPayload.value,
```

Comment:
assign a value to `maybeXmlPayload.value`

`const doc = maybeXmlPayload.value`